pkg/kubectl/describe.go

```diff
@@ -528,7 +529,30 @@ func describeContainers(pod *api.Pod, out io.Writer) {
 
 		fmt.Fprintf(out, "    Ready:\t%v\n", printBool(status.Ready))
 		fmt.Fprintf(out, "    Restart Count:\t%d\n", status.RestartCount)
+		fmt.Fprintf(out, "    Environment Variables:\t\n")
+		for _, e := range container.Env {
+			if e.ValueFrom != nil && e.ValueFrom.FieldRef != nil {
+				valueFrom := envValueFrom(pod, e)
+				fmt.Fprintf(out, "        \t%s\t = %s (%s:%s)\n", e.Name, valueFrom, e.ValueFrom.FieldRef.APIVersion, e.ValueFrom.FieldRef.FieldPath)
+			} else {
+				fmt.Fprintf(out, "        \t%s\t = %s\n", e.Name, e.Value)
+			}
+		}
+	}
+}
+
+func envValueFrom(pod *api.Pod, e api.EnvVar) string {
+	internalFieldPath, _, err := api.Scheme.ConvertFieldLabel(e.ValueFrom.FieldRef.APIVersion, "Pod", e.ValueFrom.FieldRef.FieldPath, "")
+	if err != nil {
+		return err.Error()
 	}
+
+	valueFrom, err2 := fieldpath.ExtractFieldPathAsString(pod, internalFieldPath)
```

Comment:
@krousey 
I think the pod create validation will catch creating invalid variables so it shouldn't really be triggered, I just didn't want to drop it, but it's probably pointless, so I've removed the error part.
I can add it back and handle it as if it's a possible error if you prefer

@jlowdermilk 
I am not clear on what you mean
I created a pod with  `"fieldPath": "metadata.name"` and `kubectl describe pod` has the right output
Trying to create a pod `kubectl create -f` with say  `"fieldPath": `"metadata.creationTimestamp"` is an error